aries/aries_vcx_core/src/wallet/base_wallet.rs

```diff
@@ -1,124 +1,440 @@
-use std::collections::HashMap;
-
 use async_trait::async_trait;
+use derive_builder::Builder;
+#[cfg(feature = "vdrtools_wallet")]
+use indy_api_types::domain::wallet::Record as IndyRecord;
+use public_key::Key;
+use serde::{Deserialize, Serialize};
+
+use super::entry_tag::EntryTags;
+use crate::{errors::error::VcxCoreResult, wallet::structs_io::UnpackMessageOutput};
+
+#[derive(Debug, Default, Clone, Builder)]
+pub struct Record {
+    category: String,
+    name: String,
+    value: String,
+    #[builder(default)]
+    tags: EntryTags,
+}
+
+impl Record {
+    pub fn get_value(&self) -> &str {
+        &self.value
+    }
+
+    pub fn get_name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn get_category(&self) -> &str {
+        &self.category
+    }
+
+    pub fn get_tags(&self) -> &EntryTags {
+        &self.tags
+    }
+}
+
+#[cfg(feature = "vdrtools_wallet")]
+impl From<IndyRecord> for Record {
+    fn from(ir: IndyRecord) -> Self {
+        Self {
+            name: ir.id,
+            category: ir.type_,
+            value: ir.value,
+            tags: ir.tags.into(),
+        }
+    }
+}
 
-use super::structs_io::UnpackMessageOutput;
 #[cfg(feature = "vdrtools_wallet")]
-use crate::WalletHandle;
-use crate::{errors::error::VcxCoreResult, utils::async_fn_iterator::AsyncFnIterator};
+impl From<Record> for IndyRecord {
+    fn from(record: Record) -> Self {
+        Self {
+            id: record.name,
+            type_: record.category,
+            value: record.value,
+            tags: record.tags.into(),
+        }
+    }
+}
 
-/// Trait defining standard 'wallet' related functionality. The APIs, including
-/// input and output types are loosely based off the indy Wallet API:
-/// see: <https://github.com/hyperledger/indy-sdk/blob/main/libindy/src/api/wallet.rs>
-#[async_trait]
-pub trait BaseWallet: std::fmt::Debug + Send + Sync {
-    #[cfg(feature = "vdrtools_wallet")]
-    fn get_wallet_handle(&self) -> WalletHandle;
+#[derive(Debug, Deserialize, Serialize)]
+pub struct DidData {
+    did: String,
+    verkey: Key,
+}
+
+impl DidData {
+    pub fn new(did: &str, verkey: Key) -> Self {
+        Self {
+            did: did.into(),
+            verkey,
+        }
+    }
 
-    // ----- DIDs
+    pub fn get_did(&self) -> &str {
+        &self.did
+    }
+
+    pub fn get_verkey(&self) -> &Key {
+        &self.verkey
+    }
+}
+
+pub enum SearchFilter {
+    JsonFilter(String),
+}
+
+pub trait BaseWallet: RecordWallet + DidWallet + Send + Sync + std::fmt::Debug {}
+
+#[async_trait]
+pub trait DidWallet {
     async fn create_and_store_my_did(
         &self,
         seed: Option<&str>,
-        kdf_method_name: Option<&str>,
-    ) -> VcxCoreResult<(String, String)>;
+        method_name: Option<&str>,
```

Comment:
If the type doesn't reflect what the thing is, at least name should. I am in favor of the original name here